File: src/priv/bajdcc/lexer/automata/BreadthFirstSearch.java
package priv.bajdcc.lexer.automata;

import java.util.ArrayList;

import priv.bajdcc.utility.VisitBag;

/**
 * 宽度优先搜索
 * 
 * @author bajdcc
 * @param T
 *            状态类型
 */
public class BreadthFirstSearch<Edge, Status> implements
		IBreadthFirstSearch<Edge, Status> {

	/**
	 * 存放状态的集合
	 */
	public ArrayList<Status> arrStatus = new ArrayList<Status>();

	@Override
	public boolean testEdge(Edge edge) {
		return true;
	}

	@Override
	public void visitBegin(Status status, VisitBag bag) {

	}

	@Override
	public void visitEnd(Status status) {

	}
}

File: src/main/kotlin/com/bajdcc/util/lexer/automata/BreadthFirstSearch.kt
package com.bajdcc.util.lexer.automata

import com.bajdcc.util.VisitBag

/**
 * 宽度优先搜索
 *
 * @author bajdcc 状态类型
 */
open class BreadthFirstSearch<Edge, Status> : IBreadthFirstSearch<Edge, Status> {

    /**
     * 存放状态的集合
     */
    var arrStatus = mutableListOf<Status>()

    override fun testEdge(edge: Edge): Boolean {
        return true
    }

    override fun visitBegin(status: Status, bag: VisitBag) {

    }

    override fun visitEnd(status: Status) {

    }
}
